core/src/main/java/org/mule/runtime/core/api/context/notification/NotificationHelper.java

```diff
@@ -68,13 +69,27 @@ public void fireNotification(Component source, CoreEvent event, ComponentLocatio
     try {
       if (defaultNotificationHandler.isNotificationEnabled(notificationClass)) {
         defaultNotificationHandler
-            .fireNotification(new ConnectorMessageNotification(createInfo(event, null, source), location, action));
+            .fireNotification(createConnectorMessageNotification(source, event, location, action));
       }
     } catch (Exception e) {
       logger.warn("Could not fire notification. Action: " + action, e);
     }
   }
 
+    /**
+     * Creates {@link ConnectorMessageNotification} to be fired.
+     *
+     * @param source
+     * @param event {@link CoreEvent}
+     * @param location the location of the component that generated the notification
+     * @param action the action code for the notification
+     * @return
+     */
+  public static ConnectorMessageNotification createConnectorMessageNotification(Component source, Event event,
```

Comment:
this doesn't belong there. This is an API class. This is clearly an internal thing that should live in an internal utils